client/src/app/account-card/account-card.component.ts

```diff
@@ -13,18 +14,31 @@ export class AccountCardComponent implements OnInit {
 
   public onCardClick(): void {
     this.accountService.activateAccount(this.account._id);
+    this.accountService.setAccount(this.accountService.activeAccount);
     this.transactionService.getTransactions(this.account._id).subscribe({
       next: (data) => (this.transactionService.transactions = data),
```

Comment:
I missed it previous time, it is bad to change value of service from component, if you need transactions here - add variable to this component